container_images/gce-img-resource/README.md

```diff
@@ -0,0 +1,69 @@
+# GCE Image Resource
+
+Versions [Google Compute Engine][gce] (GCE) images, either by image creation date or versioned image name.
+
+This resource is based on [frodenas/gcs-resource][frodenas], which itself is based on the official
+[S3 resource][s3-resource].
+
+## Source Configuration
+
+* `project`: *Required.* The name of the GCP project.
+
+* `family`: *Optional.* GCE Image family name. If set, versions are only produced for images with this family.
+
+* `regexp`: *Optional.* **Not yet implemented** Regular expression to apply to image names. Must contain a
+  capture group which matches a version in the image name. If provided, images are ordered by this version rather than creation date, and images which don't match or provide a valid version in the capture group will be skipped. Example:
+
+    ```yaml
+    regexp: my-image-v([0-9]+)
+    ```
+
+* `json_key`: *Optional.* **Not yet implemented** Raw JSON key to use for service account credentials. If not
+  set, uses Google Default Credentials (recommended on GCP platforms).
+
+### Example
+
+Resource configuration that produces a version for *every* GCE image published to your project (the default):
+
+```yaml
+---
+resources:
+- name: latest-image
+  type: gce-image
+  source:
+    project: my-gcp-project
+
+jobs:
+- name: my-job
+  plan:
+  - get: latest-image
+    trigger: true
+  - task: some-task
+    config: {}
+```
+
+## Behavior
+
+Principally this resource is used for representing GCE Image objects in Concourse for tracking and
+job-triggering. GCE Images don't have any downloadable content, and image creation is done by reference.
```

Comment:
no i am explaining that compared to other resources like the GCS resource, there's no meaningful concept of putting out something from concourse to a GCE image. a GCE image can be created by reference to a GCS URI or by reference to another GCE Image, Disk or Snapshot, you can't directly create it. obviously this wasn't clear enough, welcome suggestions for improving the comment :smile:

Reply:
I guess I was trying to understand whether concourse's `put` operation is supported by this resource, and it sounds like it will be, and that the caller needs to provide a reference to one of the object types listed below in `### out: Create an image.`

Reply:
we may implement `put` (called `out` in this code) if it is useful for us, or if we ever upstream this